src/config/datadog.rs

```diff
@@ -0,0 +1,117 @@
+use super::{ComponentId, Config, SinkOuter, SourceOuter};
+use crate::{
+    sinks::datadog::metrics::DatadogConfig, sources::internal_metrics::InternalMetricsConfig,
+};
+use serde::{Deserialize, Serialize};
+
+static INTERNAL_METRICS_KEY: &str = "#datadog_internal_metrics";
+static DATADOG_METRICS_KEY: &str = "#datadog_metrics";
+
+#[derive(Debug, Deserialize, Serialize, PartialEq, Clone)]
+#[serde(default, deny_unknown_fields)]
+pub struct Options {
+    #[serde(default = "default_enabled")]
+    pub enabled: bool,
+
+    #[serde(default = "default_api_key")]
+    pub api_key: Option<String>,
```

Comment:
Per https://github.com/timberio/vector/pull/8791#discussion_r692673007, I think we should avoid allowing magic variables and make interpolation explicit in config.

It feels to me that `DD_API_KEY` is likely to be present in a Datadog customer environment with the agent. While convenient to have this picked up, we'd require the customer to set `enabled: true` to turn this on anyway (to avoid surprises), so it's not any more/less convenient to have to set enabled vs. api_key.

I'd rather flip that round to the current behaviour, so that enabled is already true, and it's the API key that needs to be explicit.

Reply:
This is no longer true; `DATADOG_API_KEY` and `DD_API_KEY` will be read from the environment where an explicit `api_key` is not provided.